kbase-extension/static/kbase/js/widgets/narrative_core/kbaseDataCard.js

```diff
@@ -0,0 +1,209 @@
+            var type = entry.type;
+            var editBy = entry.editedBy ? entry.editedBy : (' by ' + object_info[5]);
+            if(!entry.type){
+                var type_tokens = object_info[2].split('.');
+                type = entry.type ? entry.type : type_tokens[1].split('-')[0];
+            }
+
+            //shorten name if applicable
+            var isShortened = false;
+            if ((entry.max_name_length && shortName) && shortName.length > entry.max_name_length) {
+                shortName = shortName.substring(0, entry.max_name_length - 3) + '...';
+                isShortened = true;
+            }
+
+            var $logo = $('<div>');
+            Icon.buildDataIcon($logo, type, entry.is_set, 0);
+                
+            var $name = $('<span>').addClass('kb-data-list-name').append(shortName);
+            var $version = $('<span>').addClass('kb-data-list-version').append(version);
+            var $type = $('<div>').addClass('kb-data-list-type').append(type);
+            var $date = $('<span>').addClass('kb-data-list-date').append(date);
+
+            //no default
+            var $byUser = $('<span>').addClass('kb-data-list-edit-by').append(editBy);
+            var $narrative = $('<div>').addClass('kb-data-list-narrative').append(entry.narrative);
+            
+            var $title = $('<div>').append($name);
+            var $subcontent = $('<div>')
+                .addClass('kb-data-list-subcontent');
+            
+            if(entry.version === undefined || entry.version) {
+                $title.append($version);
+            }
+            
+            if(entry.type === undefined || entry.type) {
+                $subcontent.append($type);
+            }
+            if(entry.narrative === undefined || entry.narrative) {
+                $subcontent.append($narrative);
+            }
+            if(entry.date === undefined || entry.date) {
+                $subcontent.append($date);
+            }
+
+            if(entry.editedBy === undefined || entry.editedBy) {
+                $byUser
+                    .click(function (object_info, e) {
+                        e.stopPropagation();
+                        window.open('/#people/' + object_info[5]);
+                    }.bind( null, object_info));
+                $subcontent.append($byUser);
+            }
+
+            //tooltip for long title
+            if (isShortened) {
+                $name.tooltip({
+                    title: entry.name,
+                    placement: 'bottom',
+                    delay: {
+                        show: Config.get('tooltip').showDelay,
+                        hide: Config.get('tooltip').hideDelay
+                    }
+                });
+            }
+            
+            //create card
+            var actionButtonClick = function (e) {
+                if(!entry.ws_name){
+                    return;
+                }
+                e.preventPropagation; 
```

Comment:
This should be a function call - `e.preventDefault()` or `e.stopPropagation()` there's no `preventPropagation`

Reply:
ops typod. supprised it worked